# Task 1

In [20]:
import pandas as pd
df=pd.read_csv("house-prices-advanced-regression-techniques/train.csv")
print("shape of dataset : ",df.shape)
print("\nDatatypes od dataset :\n",df.dtypes)
print("\nFirst few rows of dataset :\n",df.head(10))
print("\nSummary statistics of dataset :\n",df.describe())

shape of dataset :  (1460, 81)

Datatypes od dataset :
 Id                 int64
MSSubClass         int64
MSZoning             str
LotFrontage      float64
LotArea            int64
                  ...   
MoSold             int64
YrSold             int64
SaleType             str
SaleCondition        str
SalePrice          int64
Length: 81, dtype: object

First few rows of dataset :
    Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   
5   6          50       RL         85.0    14115   Pave   NaN      IR1   
6   7          20       RL         75.0    10084   Pave   NaN      Reg   
7   8          60    

# Task 2

In [21]:
y = df["SalePrice"]
X = df.drop(columns=["SalePrice", "Id"])
numerical_features=X.select_dtypes(include=["int64","float64"]).columns
categorical_features=X.select_dtypes(include=["str"]).columns
print("Numerical features :\n",numerical_features)
print("\nCategorical features :\n",categorical_features)

Numerical features :
 Index(['MSSubClass', 'LotFrontage', 'LotArea', 'OverallQual', 'OverallCond',
       'YearBuilt', 'YearRemodAdd', 'MasVnrArea', 'BsmtFinSF1', 'BsmtFinSF2',
       'BsmtUnfSF', 'TotalBsmtSF', '1stFlrSF', '2ndFlrSF', 'LowQualFinSF',
       'GrLivArea', 'BsmtFullBath', 'BsmtHalfBath', 'FullBath', 'HalfBath',
       'BedroomAbvGr', 'KitchenAbvGr', 'TotRmsAbvGrd', 'Fireplaces',
       'GarageYrBlt', 'GarageCars', 'GarageArea', 'WoodDeckSF', 'OpenPorchSF',
       'EnclosedPorch', '3SsnPorch', 'ScreenPorch', 'PoolArea', 'MiscVal',
       'MoSold', 'YrSold'],
      dtype='str')

Categorical features :
 Index(['MSZoning', 'Street', 'Alley', 'LotShape', 'LandContour', 'Utilities',
       'LotConfig', 'LandSlope', 'Neighborhood', 'Condition1', 'Condition2',
       'BldgType', 'HouseStyle', 'RoofStyle', 'RoofMatl', 'Exterior1st',
       'Exterior2nd', 'MasVnrType', 'ExterQual', 'ExterCond', 'Foundation',
       'BsmtQual', 'BsmtCond', 'BsmtExposure', 'BsmtFinType1', 'BsmtFinTy

# Task 2 interpretation
Dependent variable:
SalePrice:the price of the house that we want to predict.
Independent variables:
All other relevant columns in the dataset are independent variables.

# Task 3
This is a regression problem because the target variable SalePrice is a continuous numerical value that we want to predict. Therefore, the model will be evaluated using regression metrics such as MAE, MSE/RMSE, and R² rather than classification metrics such as accuracy or precision.

# Part 2: Data Preparation & Train/Test Split 

# Task 4

In [24]:
selected_features = [
    "GrLivArea",
    "OverallQual",
    "TotalBsmtSF",
    "GarageCars",
    "YearBuilt",
    "FullBath",
    "GarageArea",
    "1stFlrSF",
    "Neighborhood",
    "HouseStyle",
    "KitchenQual",
    "ExterQual"
]
X = df[selected_features].copy()
y = df["SalePrice"]
categorical_features = X.select_dtypes(include=["str"]).columns
X=pd.get_dummies(X,columns=categorical_features,drop_first=True)
print(X.shape)

(1460, 45)


# Task 4 interpretation
Categorical features were converted into numerical form using one-hot encoding because the categories do not have an inherent order. Numerical features were kept in their original form. SalePrice was used as the target variable, while Id was removed because it is only an identifier and does not provide meaningful information for predicting house prices.


# Task 5

In [25]:
X = X.fillna(X.median(numeric_only=True))
X = X.astype(int)
print("Data types:")
print(X.dtypes.value_counts())
print("\nTotal missing values:")
print(X.isnull().sum().sum())

Data types:
int64    45
Name: count, dtype: int64

Total missing values:
0


# Task 5 interpretation
After encoding the categorical features and handling missing values, all 45 input features are in numerical form. The dataset contains 0 missing values, so the feature matrix X is ready to be used for training the Linear Regression model.

# Task 6

In [27]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X, y,
    test_size=0.2,
    random_state=42
)

print("X_train shape:", X_train.shape)
print("X_test shape:", X_test.shape)
print("y_train shape:", y_train.shape)
print("y_test shape:", y_test.shape)

X_train shape: (1168, 45)
X_test shape: (292, 45)
y_train shape: (1168,)
y_test shape: (292,)


# Task 6 interpretation
The dataset was divided into 80% training data and 20% testing data using train_test_split() with random_state=42. The training set contains 1,168 houses, while the test set contains 292 houses, with 45 features in each set.